dcpy/models/product/dataset/metadata_v2.py

```diff
@@ -0,0 +1,424 @@
+from __future__ import annotations
+
+import jinja2
+
+from pathlib import Path
+from pydantic import BaseModel, field_validator, model_serializer
+from pydantic.fields import PrivateAttr
+from typing import Any, List, Literal, get_args, Self
+import typing
+import unicodedata
+import yaml
+
+from dcpy.utils.logging import logger
+from dcpy.utils.collections import deep_merge_dict as merge
+
+
+# MISC UTILS
+def normalize_text(s):
+    """
+    Normalize the text we may receive from the various metadata sources.
+    Primarily useful for cleaning long-text like descriptions.
+    """
+    char_map = {
+        "–": "-",  # en dash to hyphen
+        "—": "-",  # em dash to hyphen
+        "’": "'",  # curly apostrophe
+        "“": '"',  # lcurly quote
+        "”": '"',  # rcurly quote
+    }
+    translator = str.maketrans(char_map)
+    # Normalize Unicode characters
+    normalized = unicodedata.normalize("NFKD", s)
+    # Apply the translation
+    cleaned = normalized.translate(translator)
+    return cleaned.strip()
+
+
+# BASE
+class SortedSerializedBase(BaseModel):
+    """A Pydantic BaseModel that will allow for sensible (and overrideable) deserialization order.
+
+    The serialization order is as follows:
+    - model attributes defined in the head sort order
+    - simple (and nullable simple) types: strings, ints, bools, literals
+    - complex types
+    - model attributes defined in the tail sort order
```

Comment:
I think I like how this works out, in terms what the output looks like. 

Is there a reason you went this way rather than, say, something like trying to serialize in the order the model is actually defined?